### ISSUE TITLE
Grouped Conv1d

### ISSUE THREAD

**Issue by mssdot** [inokoj/VAP-Realtime#9]
Hi, I have a question regarding the use of Conv1D layers in encoder, why used normal Conv1ds typically used instead of grouped convolutions? grouped convolutions can significantly reduce the number of parameters and MACs, so is there a specific reason they are not commonly used in this context?

if not ,you could consider trying grouped convolutions—they might help with efficiency in Realtime without sacrificing performance too much

**Comment by inokoj** [comment]
Thank you. We just used the original model without modifications.

Are you referring to something like this?
In CPCEncoder:
`nn.Conv1d(sizeHidden, sizeHidden, 4, stride=2, padding=1, groups=num_groups)`

Implementing this change would require retraining the CPC model itself as part of a pretraining step.
While this is feasible, it may take some time.
